DNN Platform/Library/Entities/Portals/PortalController.cs

```diff
@@ -163,6 +163,12 @@ private static void CreateDefaultPortalRoles(int portalId, int administratorId,
             }
             string mappedHomeDirectory = String.Format(Globals.ApplicationMapPath + "\\" + homeDirectory + "\\").Replace("/", "\\");
 
+            if(Directory.Exists(mappedHomeDirectory))

```

Comment:
Maybe this check should actually be done by reading `HomeDirectory` column from `Portals` table? @dnnsoftware/tag

Reply:
Now that I think about it, I would even enforce uniqueness of the home directory. I think it makes no sense for multiple portals to share one and would have strange side-effects

Reply:
On third thought, checking the directory does also make sense. 

Question is what should happen if no portal exists in the database using a given directory but it exists in filesystem. It should not happen but could be possible I expect. Would it be fine to overwrite then?

Reply:
With regards to your third thought, the way this would happen is a deleted portal that didn't delete cleanly with a custom home directory